doc/introduction/data.rst

```diff
@@ -0,0 +1,162 @@
+.. role:: html(raw)
+   :format: html
+
+.. _intro_ref_data:
+
+Quantum Datasets
```

Comment:
Please double check the code examples one more time. Majority of the examples do not work properly.